File: split/newnullbox.pas
function newnullbox: halfword;
var
  p: halfword;
begin
  p := getnode(7);
  mem[p].hh.b0 := 0;
  mem[p].hh.b1 := 0;
  mem[p + 1].int := 0;
  mem[p + 2].int := 0;
  mem[p + 3].int := 0;
  mem[p + 4].int := 0;
  mem[p + 5].hh.rh := 0;
  mem[p + 5].hh.b0 := 0;
  mem[p + 5].hh.b1 := 0;
  mem[p + 6].gr := 0.0;
  newnullbox := p;
end;

File: cpp/newnullbox.cpp
#include "newnullbox.h"
#include "getnode.h"

halfword newnullbox(void)
{
	halfword p = getnode(box_node_size);
	type(p) = hlist_node;
	subtype(p) = 0;
	width(p) = 0;
	depth(p) = 0;
	height(p) = 0;
	shift_amount(p) = 0;
	list_ptr(p) = 0;
	glue_sign(p) = normal;
	glue_order(p) = normal;
	glue_set(p) = 0.0;
	return p;
}
